packages/core/test/lib/utils/traceData.test.ts

```diff
@@ -1,156 +1,280 @@
-import { SentrySpan, getTraceData } from '../../../src/';
-import * as SentryCoreCurrentScopes from '../../../src/currentScopes';
-import * as SentryCoreExports from '../../../src/exports';
-import * as SentryCoreTracing from '../../../src/tracing';
-import * as SentryCoreSpanUtils from '../../../src/utils/spanUtils';
+import type { Client, Span } from '@sentry/types';
+import {
+  SentrySpan,
+  getCurrentScope,
+  getGlobalScope,
+  getIsolationScope,
+  getMainCarrier,
+  getTraceData,
+  setAsyncContextStrategy,
+  setCurrentClient,
+  withActiveSpan,
+} from '../../../src/';
+import { getAsyncContextStrategy } from '../../../src/asyncContext';
+import { freezeDscOnSpan } from '../../../src/tracing/dynamicSamplingContext';
 
 import { isValidBaggageString } from '../../../src/utils/traceData';
+import type { TestClientOptions } from '../../mocks/client';
+import { TestClient, getDefaultTestClientOptions } from '../../mocks/client';
 
-const TRACE_FLAG_SAMPLED = 1;
+const dsn = 'https://123@sentry.io/42';
 
-const mockedSpan = new SentrySpan({
-  traceId: '12345678901234567890123456789012',
-  spanId: '1234567890123456',
-  sampled: true,
-});
+const SCOPE_TRACE_ID = '12345678901234567890123456789012';
+const SCOPE_SPAN_ID = '1234567890123456';
 
-const mockedClient = {} as any;
+function setupClient(opts?: Partial<TestClientOptions>): Client {
+  getCurrentScope().setPropagationContext({
+    traceId: SCOPE_TRACE_ID,
+    spanId: SCOPE_SPAN_ID,
+  });
 
-const mockedScope = {
-  getPropagationContext: () => ({
-    traceId: '123',
-  }),
-} as any;
+  const options = getDefaultTestClientOptions({
+    dsn,
+    tracesSampleRate: 1,
+    ...opts,
+  });
+  const client = new TestClient(options);
+  setCurrentClient(client);
+  client.init();
+
+  return client;
+}
 
 describe('getTraceData', () => {
```

Comment:
these tests relied a lot on mocks, and since the underlying implementation was changed a bunch of this broke. So I rewrote these tests to use "real" stuff as much as possible, but tried to keep the scenarios we want to cover intact.